### PR TITLE
EVG-7099 Add ability to limit distros to projects

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

**Comment by bsamek** [comment]
@tychoish did the thing

File: service/templates/distros.html [hunk 1]
```diff
@@ -887,6 +887,11 @@ <h2 style="display:inline-block; padding-right:15px">Configure
                 Expansion</button>
             </div>
           </div>
+          <div>
+            <label class="distro-label">Valid Projects:</label>
+            <textarea ng-readonly="readOnly" name="valid_projects" ng-list="&#10;" class="form-control" rows="5"
```
Comment (john-m-liu) [inline service/templates/distros.html]: This needs ng-trim="false" otherwise each character will be considered a separate project

Comment (bsamek) [reply service/templates/distros.html]: done

File: model/task/task.go [hunk 2]
```diff
@@ -1444,14 +1445,24 @@ func FindSchedulableForAlias(id string) ([]Task, error) {
 
 func FindRunnable(distroID string, removeDeps bool) ([]Task, error) {
 	match := scheduleableTasksQuery()
+	var d distro.Distro
+	var err error
 	if distroID != "" {
 		match[DistroIdKey] = distroID
+		d, _ = distro.FindOne(distro.ById(distroID).WithFields(distro.ValidProjectsKey))
```
Comment (ablack12) [inline model/task/task.go]: You're ignoring the error here but then checking it

File: rest/model/distro.go [hunk 3]
```diff
@@ -354,6 +354,7 @@ type APIDistro struct {
 	DisableShallowClone   bool                     `json:"disable_shallow_clone"`
 	UseLegacyAgent        bool                     `json:"use_legacy_agent"`
 	Note                  APIString                `json:"note"`
+	ValidProjects         []string                 `json:"valid_projects"`
```
Comment (ablack12) [inline rest/model/distro.go]: I don't know the pros and cons, but there's a function ToAPIStringList (and From) if you want to do that

Comment (bsamek) [reply rest/model/distro.go]: ah, didn't realize, fixed

File: model/task/task.go (hunk 2 above)
Comment (bsamek) [reply model/task/task.go]: fixed

**Pull request merged**